### PR TITLE
Adding multiple data sources

### PR DESCRIPTION
**Pull request by steweg**
This patch adds the following data sources:
- azurerm_virtual_hub_route_table
- azurerm_marketplace_agreement

### CODE REVIEW

File: internal/services/compute/marketplace_agreement_data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+)
+
+type MarketplaceAgreementDataSource struct{}
+
+func TestAccDataSourceMarketplaceAgreement_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_marketplace_agreement", "test")
+	r := MarketplaceAgreementDataSource{}
+	offer := "waf"
```
[the remark this answers is not in the document]
Comment (steweg) [reply internal/services/compute/marketplace_agreement_data_source_test.go]: Ok, I have adjusted it to be hardcoded inside the test

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @steweg 

Thanks for pushing those changes, I've taken a look through and left a few more minor comments inline, if we can fix those up then these two should be good to merge 👍 

Thanks!

File: internal/services/compute/marketplace_agreement_data_source.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+	id := parse.NewPlanID(subscriptionId, d.Get("publisher").(string), d.Get("offer").(string), d.Get("plan").(string))
+
+	log.Printf("[DEBUG] retrieving %s", id)
+
+	term, err := client.Get(ctx, id.AgreementName, id.OfferName, id.Name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(term.Response) {
+			return fmt.Errorf("retrieving %s: %s", id, err)
+		}
```
[the remark this answers is not in the document]
Comment (steweg) [reply internal/services/compute/marketplace_agreement_data_source.go]: Done

File: internal/services/network/virtual_hub_route_table_data_source.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+			"labels": {
+				Type:     pluginsdk.TypeSet,
+				Computed: true,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+				},
+			},
+
+			"route": {
+				Type:     pluginsdk.TypeSet,
```
[the remark this answers is not in the document]
Comment (steweg) [reply internal/services/network/virtual_hub_route_table_data_source.go]: Done - but here I have followed the schema within the existing "resource". So I guess it should be fixed in the resource as well, which is beyond scope of this patch.

File: internal/services/network/virtual_hub_route_table_data_source.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"destinations": {
+							Type:     pluginsdk.TypeSet,
```
[the remark this answers is not in the document]
Comment (steweg) [reply internal/services/network/virtual_hub_route_table_data_source.go]: Done - the same comment about the resource as above

File: internal/services/network/virtual_hub_route_table_data_source.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	defer cancel()
+
+	id := parse.NewHubRouteTableID(subscriptionId, d.Get("resource_group_name").(string), d.Get("virtual_hub_name").(string), d.Get("name").(string))
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("%s was not found", id)
+		}
+		return fmt.Errorf("reading %s: %+v", id, err)
```
[the remark this answers is not in the document]
Comment (steweg) [reply internal/services/network/virtual_hub_route_table_data_source.go]: Done

File: website/docs/d/virtual_hub_route_table.html.markdown [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+The following attributes are exported:
+
+* `id` - The ID of the Virtual Hub Route Table.
+
+* `virtual_hub_id` - The ID of the Virtual Hub within which this route table is created
+
+* `labels` - List of labels associated with this route table.
+
+* `route` - A `route` block as defined below.
```
[the remark this answers is not in the document]
Comment (steweg) [reply website/docs/d/virtual_hub_route_table.html.markdown:44]: Done

File: internal/services/network/virtual_hub_route_table_data_source.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (katbyte) [reply internal/services/network/virtual_hub_route_table_data_source.go]: data sources are always lists regardless of if the resources is a set or not so all good here :)

**Comment by steweg** [comment]
@tombuildsstuff As all the open items were resolved, is there anything else needed from my side, or is it ready to be merged? Thanks

**Review by katbyte — APPROVED** [review]
LGTM 🦀

**Pull request merged**